pkg/reconciler/workload/synctarget/synctarget_controller.go

```diff
@@ -191,11 +191,23 @@ func (c *Controller) process(ctx context.Context, key string) error {
 		return err
 	}
 
-	if _, err := c.kcpClusterClient.WorkloadV1alpha1().SyncTargets().Patch(logicalcluster.WithCluster(ctx, logicalcluster.From(currentSyncTarget)), currentSyncTarget.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{}, "status"); err != nil {
-		klog.Errorf("failed to patch sync target status: %v", err)
-		return err
+	if !reflect.DeepEqual(currentSyncTarget.ObjectMeta, newSyncTarget.ObjectMeta) || !reflect.DeepEqual(currentSyncTarget.Spec, newSyncTarget.Spec) {
+		if _, err := c.kcpClusterClient.WorkloadV1alpha1().SyncTargets().Patch(logicalcluster.WithCluster(ctx, logicalcluster.From(currentSyncTarget)), currentSyncTarget.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{}); err != nil {
+			klog.Errorf("failed to patch sync target: %v", err)
+			return err
+		}
+		klog.V(4).InfoS("patched synctarget", "name", newSyncTarget.Name, "workspace", logicalcluster.From(newSyncTarget))
```

Comment:
nit: would do that before the patch and include the actual patch bytes, and V(2).

Reply:
done